CMakeLists.txt

```diff
@@ -1,94 +1,242 @@
-option(KOMPUTE_OPT_BUILD_DOCS "Enable if you want to build documentation" 0)
-option(KOMPUTE_OPT_BUILD_SHADERS "Enable if you want to re-build all shader files" 0)
-option(KOMPUTE_OPT_BUILD_SINGLE_HEADER "Enable if you want to build the single header file" 0)
-option(KOMPUTE_OPT_INSTALL "Enable if you want to enable installation" 0)
+kompute_option(KOMPUTE_OPT_BUILD_TESTS "Enable if you want to build tests." OFF)
+kompute_option(KOMPUTE_OPT_CODE_COVERAGE "Enable if you want code coverage." OFF)
+kompute_option(KOMPUTE_OPT_BUILD_DOCS "Enable if you want to build documentation." OFF)
+kompute_option(KOMPUTE_OPT_INSTALL "Enable if you want to enable installation." OFF)
+
 # Build options
-option(KOMPUTE_OPT_BUILD_PYTHON "Enable if you want to build python bindings" 0)
-option(KOMPUTE_OPT_ENABLE_SPDLOG "Enable to compile with spdlog as the internal logging framework" 0)
-option(KOMPUTE_OPT_REPO_SUBMODULE_BUILD "Use the submodule repos instead of external package manager" 0)
-option(KOMPUTE_OPT_ANDROID_BUILD "Enable android compilation flags required" 0)
-option(KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS "Explicitly disable debug layers even on debug" 0)
-option(KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS "Whether to use shared libraries for dependencies for install" 0)
-option(KOMPUTE_OPT_BUILD_AS_SHARED_LIB "Whether to build kompute as shared library" 0)
-# Build flags
-set(KOMPUTE_EXTRA_CXX_FLAGS "" CACHE STRING "Extra compile flags for Kompute, see docs for full list")
-
-if(KOMPUTE_OPT_ENABLE_SPDLOG)
-    set(KOMPUTE_EXTRA_CXX_FLAGS "${KOMPUTE_EXTRA_CXX_FLAGS} -DKOMPUTE_ENABLE_SPDLOG=1")
-    set(SPDLOG_FMT_EXTERNAL ON CACHE BOOL "Enables external fmt as its current dep" FORCE)
-    if(KOMPUTE_OPT_INSTALL)
-        # Enable install parameters for spdlog (overrides parameters passed)
-        set(SPDLOG_INSTALL ON CACHE BOOL "Enables install of spdlog" FORCE)
-
-        if(KOMPUTE_OPT_DEPENDENCIES_SHARED_LIBS)
-            set(SPDLOG_BUILD_SHARED ON CACHE BOOL "Enables build of shared libraries" FORCE)
-        endif()
+kompute_option(KOMPUTE_OPT_BUILD_PYTHON "Enable if you want to build python bindings." OFF)
+kompute_log_level(KOMPUTE_OPT_LOG_LEVEL "Internally we use spdlog for logging. The log level used can be changed here. Possible values: 'Trace', 'Debug', 'Info', 'Warn', 'Error', 'Critical', 'Off', 'Default'. If set to 'Off' spdlog will be deactivated completely. If set to 'Default', the log level will be set to 'Info' for release builds and 'Debug' else." "Default")
+kompute_option(KOMPUTE_OPT_ANDROID_BUILD "Enable android compilation flags required." OFF)
+kompute_option(KOMPUTE_OPT_DISABLE_VK_DEBUG_LAYERS "Explicitly disable debug layers even on debug." OFF)
+kompute_option(KOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK "Whether to check if your driver supports the Vulkan Header version you are linking against. This might be useful in case you build shared on a different system than you run later." OFF)
```

Comment:
MOving the comments made into here:

Just tried from the latest to build but it seems it gets frozen for a very long time here:

```
...etc
--   KOMPUTE_OPT_BUILD_SHADERS: 1
--   KOMPUTE_OPT_USE_BUILD_IN_SPDLOG: ON
--   KOMPUTE_OPT_USE_BUILD_IN_FMT: ON
--   KOMPUTE_OPT_USE_BUILD_IN_GOOGLE_TEST: ON
--   KOMPUTE_OPT_USE_BUILD_IN_PYBIND11: ON
--   KOMPUTE_OPT_USE_BUILD_IN_VULKAN_HEADER: ON
--   KOMPUTE_OPT_BUILD_IN_VULKAN_HEADER_TAG: v1.2.203
-- =======================================================
-- Ensuring the currently installed driver supports the Vulkan version requested by the Vulkan Header.
-- Found Vulkan Header version 1.2.203.
```

It after a while it fails. It seems to be if the DISPLAY is not set up it won't work as vulkaninfo requires it. I don't think this is something we can assume people will actually be running. I assume this is being requrested due to vulkaninfo or similar command being run correct? Would there be a way to validate version in different way? Alternatively perhaps making vulkan version check optional and highlighting that needs DISPLAY to be configured. Below is the error that shows the issue with vulkaninfo.

```
--   KOMPUTE_OPT_BUILD_IN_VULKAN_HEADER_TAG: v1.2.203
-- =======================================================
-- Ensuring the currently installed driver supports the Vulkan version requested by the Vulkan Header.
-- Found Vulkan Header version 1.2.203.
XCB failed to connect to the X server due to error:1.
ERROR at /build/vulkan-tools-1.2.148.0-1ubuntu18.04/vulkaninfo/vulkaninfo.h:822: AppCreateXcbSurface failed to establish connection
CMake Error at cmake/check_vulkan_version.cmake:63 (message):
  No GPU supporting Vulkan found in vulkaninfo.  Does your GPU (driver)
  support Vulkan?
Call Stack (most recent call first):
  CMakeLists.txt:132 (check_vulkan_version)


-- Configuring incomplete, errors o
```

Reply:
There is already an option to disable the Vulkan version check: `DKOMPUTE_OPT_DISABLE_VULKAN_VERSION_CHECK`
I further improved the Vulkan version check even further and added more instruction:
* https://github.com/COM8/kompute/commit/4895f2f9324f50d1f1d167dc98f703d2f29f9151
* https://github.com/COM8/kompute/commit/383326215ba7d50a913445eae5accf91eb55ff19

This should make it clearer what has to be done to compile Kompute.